DSharpPlus/Clients/DiscordClient.cs

```diff
@@ -835,6 +835,91 @@ public async Task<DiscordApplicationCommand> EditGuildApplicationCommandAsync(ul
+    /// </summary>
+    /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
+    /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+    public IAsyncEnumerable<DiscordGuild> GetGuildsBeforeAsync(ulong before, int limit = 200, bool? withCount = null)
+        => this.GetGuildsInternalAsync(limit, before, null, null);
+
+    /// <summary>
+    /// Returns a list of guilds after a certain guild.
+    /// <param name="limit">The amount of guilds to fetch.</param>
+    /// <param name="after">guild to fetch after from.</param>
+    /// <param name="withCount">Whether to include approximate member and presence counts in the returned guilds.</param>
+    /// </summary>
+    /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
+    /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+    public IAsyncEnumerable<DiscordGuild> GetGuildsAfterAsync(ulong after, int limit = 100, bool? withCount = null)
+        => this.GetGuildsInternalAsync(limit, null, after, null);
+    
+    /// <summary>
+    /// Returns a list of guilds the bot is in.
+    /// <param name="limit">The amount of guilds to fetch.</param>
+    /// <param name="withCount">Whether to include approximate member and presence counts in the returned guilds.</param>
+    /// </summary>
+    /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
+    /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+    public IAsyncEnumerable<DiscordGuild> GetGuildsAsync(int limit = 100, bool? withCount = null) =>
+        this.GetGuildsInternalAsync(limit, null, null, withCount);
+    
+    private async IAsyncEnumerable<DiscordGuild> GetGuildsInternalAsync(int limit = 200, ulong? before = null, ulong? after = null, bool? withCount = null)
+    {
+        if (limit < 0)
+        {
+            throw new ArgumentException("Cannot get a negative number of guilds.");
+        }
+
+        if (limit == 0)
+        {
+            yield break;
+        }
+        
+        List<DiscordGuild> guilds = new(limit);
+        int remaining = limit;
+        ulong? last = null;
+        bool isbefore = before != null;
+
+        int lastCount;
+        do
+        {
+            int fetchSize = remaining > 100 ? 100 : remaining;
+            IReadOnlyList<DiscordGuild> fetchedGuilds = await this.ApiClient.GetGuildsAsync( fetchSize, isbefore ? last ?? before : null, !isbefore ? last ?? after : null, withCount);
+
+            lastCount = fetchedGuilds.Count;
+            remaining -= lastCount;
+            
+            //We sort the returned guilds by ID so that they are in order in case Discord switches the order AGAIN.
```

Comment:
are we sure this is needed

Reply:
Im not really sure. They switched the order on the messages endpoint sometimes